runtime/vm/vmprops.c

```diff
@@ -572,39 +573,35 @@ initializeSystemProperties(J9JavaVM * vm)
 		}
 	}
 
-#if JAVA_SPEC_VERSION < 12
-	/* Following system properties are defined via java.lang.VersionProps.init(systemProperties) and following settings within System.ensureProperties() */
-	{
+	if (JAVA_SPEC_VERSION == 8) {
+		specificationVersion = "1.8";
+	} else {
+#define J9_STR_(x) #x
+#define J9_STR(x) J9_STR_(x)
+		specificationVersion = J9_STR(JAVA_SPEC_VERSION);
+	}
+	rc = addSystemProperty(vm, "java.specification.version", specificationVersion, 0);
```

Comment:
This is set by `VersionProps.init()`which the `java.vm.specification.version` is not.  To me, it makes sense for the VM to own setting the vm specific one, and the class libraries to own the other.

Reply:
Since it used to be set in the VM prior to Java 12, I suggesting we should be adding this for Java 12 and up, even though VersionProps.init() sets it later. Then the value is available in JVMTI.

Reply:
Agreed, updated PR to set `java.vm.specification.version` but not `java.specification.version` in native for `JDK12+`.

Reply:
Oops, missed Pete's comments above (no refresh).

Reply:
Doesn't this present a consistency issue?  The VM sets something early, its read in jvmti, and then VersionProps sets something subtly different at runtime. 

I would prefer to have the VM own the `*.vm.*` properties and let the class library own the others.  This seems to be inline with the design of VersionProps

Reply:
@DanHeidinga @pshipton the PR has been updated accordingly and ready for another reviewing.